sdk-src/src/main/java/com/hpe/adm/nga/sdk/Octane.java

```diff
@@ -130,13 +161,16 @@ public AttachmentList attachmentList() {
      * @return base domain
      */
     private String getBaseDomainFormat() {
-
-        String baseDomain = "";
+        String baseDomain = String.format(SHARED_SPACES_DOMAIN_FORMAT, urlDomain);
         if (idsharedSpaceId != null && !idsharedSpaceId.isEmpty()) {
-            baseDomain = String.format(SHARED_SPACES_DOMAIN_FORMAT, urlDomain, idsharedSpaceId);
-
-            if (workSpaceId != 0)
-                baseDomain = baseDomain + String.format(WORKSPACES_DOMAIN_FORMAT, workSpaceId);
+            baseDomain = String.format(ID_FORMAT, baseDomain, idsharedSpaceId);
+            if (workSpaceId == NO_WORKSPACE_ID) {
+                baseDomain = String.format(WORKSPACES_DOMAIN_FORMAT, baseDomain).concat("/");
+            } else if (workSpaceId != ONLY_SHAREDSPACE_WORKSPACE_ID) {
+                baseDomain = String.format(ID_FORMAT, String.format(WORKSPACES_DOMAIN_FORMAT, baseDomain), workSpaceId).concat("/");
+            } else {
+                baseDomain = baseDomain.concat("/");
+            }
```

Comment:
These ifs were a little confusing. I would rename "ONLY_SHAREDSPACE_WORKSPACE_ID" to "NO_WORKSPACE_DOMAIN" or something similar since it doesn't rally do anything related to the shared space. With that rename in mind, I think a flow which would be easier to understand would be something like this (please check to see if it is correct):
```
if (workSpaceId != NO_WORKSPACE_DOMAIN) {
    baseDomain = String.format(WORKSPACES_DOMAIN_FORMAT, baseDomain);
    if (workSpaceId != NO_WORKSPACE_ID) {
        baseDomain = String.format(ID_FORMAT, baseDomain, workSpaceId);
    }
}
baseDomain = baseDomain.concat("/");```

Reply:
I'll have to see if this works :)